h2/src/main/org/h2/util/ToDate.java

```diff
@@ -0,0 +1,63 @@
+package org.h2.util;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+import org.h2.util.ToDateTokenizer.FormatTokenEnum;
+
+/**
+ * Emulates Oracle's TO_DATE function.<br>
+ * Main class
+ */
+public class ToDate {
+    public enum ToDateFunctionName {
+        TO_DATE("DD MON YYYY"), TO_TIMESTAMP("DD MON YYYY HH:MI:SS");
+
+        private final String defaultFormatStr;
+
+        ToDateFunctionName(final String defaultFormatStr) {
+            this.defaultFormatStr = defaultFormatStr;
+        }
+
+        String getDefaultFormatStr() {
+            return defaultFormatStr;
+        }
+    };
```

Comment:
I see your point. The enum needs to be private and the naming is wrong. 
Should be called something like ParseType (and not ToDateFunctionName). It's used to set some initial stuff and tell the parser what type of parsing we are doing (TO_DATE of TO_TIMESTAMP) as the two methods share 95% of the code.